File: 2026-02/src/day_23/boj_35295/Main.java
package day_23.boj_35295;

import java.io.*;
import java.util.*;

public class Main {

    static final int MAX = 1000;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        boolean[] isPrime = sieve();

        int n = Integer.parseInt(br.readLine());

        int[] arr = new int[n];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(arr);

        int x = arr[n - 2];
        int y = arr[n - 1];
        if (x == 1 && isPrime[y]) {
            System.out.println("NO");
        } else {
            System.out.println("YES");
            System.out.println(2);
            System.out.println(x + " " + y);
        }
    }

    static boolean[] sieve() {
        boolean[] isPrime = new boolean[1 + MAX];
        Arrays.fill(isPrime, true);
        isPrime[0] = isPrime[1] = false;

        for (int i = 2; i * i <= MAX; i++) {
            if (isPrime[i]) {
                for (int j = i * i; j <= MAX; j += i) {
                    isPrime[j] = false;
                }
            }
        }

        return isPrime;
    }
}

File: 2026-02/src/day_23/boj_35295/Main.cpp
#include <bits/stdc++.h>
using namespace std;

const int MAX = 1000;
bool isPrime[1 + MAX];

void sieve() {
    fill(isPrime, isPrime + MAX + 1, true);
    isPrime[0] = isPrime[1] = false;

    for (int i = 2; i * i <= MAX; i++) {
        if (isPrime[i]) {
            for (int j = i * i; j <= MAX; j += i) {
                isPrime[j] = false;
            }
        }
    }
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n;
    cin >> n;

    vector<int> v(n);
    for (int& x : v) cin >> x;
    sort(v.begin(), v.end());

    sieve();

    int x = v[n - 2];
    int y = v[n - 1];
    if (x == 1 && isPrime[y]) {
        cout << "NO";
    } else {
        cout << "YES\n";
        cout << "2\n";
        cout << x << ' ' << y;
    }
}
